src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/VariantsToTable.java

```diff
@@ -278,13 +333,15 @@ private void emitMoltenizedOutput(final List<String> record) {
      * @param vc                the VariantContext whose field values we can to capture
      * @return List of lists of field values
      */
-    private List<List<String>> extractFields(final VariantContext vc) {
+    protected List<List<String>> extractFields(final VariantContext vc) {
 
         final int numRecordsToProduce = splitMultiAllelic ? vc.getAlternateAlleles().size() : 1;
         final List<List<String>> records = new ArrayList<>(numRecordsToProduce);
 
         final int numFields;
-        final boolean addGenotypeFields = genotypeFieldsToTake != null && !genotypeFieldsToTake.isEmpty();
+        final boolean hasGFields = genotypeFieldsToTake != null && !genotypeFieldsToTake.isEmpty();
```

Comment:
`genotypeFiledsToTake` defaults to an empty array, so you don't need to check that it's not null. I would do away with `hasGFields` and `hasASGFields` and just say `final boolean addGenotypeFields = !genotypeFieldsToTake.isEmpty() || !asGenotypeFieldsToTake.isEmpty();`